ssh/tailssh/incubator.go

```diff
@@ -370,26 +373,26 @@ func dropPrivileges(logf logger.Logf, wantUid, wantGid int, supplementaryGroups
 	// cannot reset the it back to our original values, and that the
 	// current egid/euid are the expected values after we change
 	// everything; if not, we exit the process.
-	if assertDropPrivileges {
+	if checkUnDropPrivileges {
 		if egid != wantGid {
 			if err := syscall.Setegid(egid); err == nil {
-				fatalf("unexpectedly able to set egid back to %d", egid)
+				fatalf("able to set egid back to %d", egid)
 			}
 		}
 		if euid != wantUid {
 			if err := syscall.Seteuid(euid); err == nil {
-				fatalf("unexpectedly able to set euid back to %d", euid)
+				fatalf("able to set euid back to %d", euid)
 			}
 		}
-
+	}
+	// TODO(andrew): remove this envknob after sufficient testing, e.g. the 1.40 release
+	if !envknob.BoolDefaultTrue("TS_DEBUG_DISABLE_ASSERT_PRIVILEGES") {
```

Comment:
there are too many negations here. Whats the intent?

Reply:
Heh, enough that I actually screwed it up 🤦 Lemme fix.

Reply:
Okay, removed a bunch of negations; PTAL?